### PR TITLE
Refactor License construction to permit loading from provided bytes.

### PR DESCRIPTION
**Pull request by dsymonds**
This permits using this package in environments without a GOPATH or
without requiring filesystem operations (e.g. by embedding licenses.db).

Also remove unnecessary mutex and loop counter from registerLicenses.

### CODE REVIEW

File: classifier.go [hunk 1]
```diff
@@ -77,29 +76,38 @@ type License struct {
 	Threshold float64
 }
 
-// New creates a license classifier and pre-loads it with known open source licenses.
-func New(threshold float64) (*License, error) {
+// NewWithDB creates a lincenses classifier based on a provided database,
```
Comment (gmlewis) [inline classifier.go]: s/lincenses/license/

Comment (dsymonds) [reply classifier.go]: done!

**Comment by gwelymernans** [comment]
The CI tests against Go 1.8 failed (but passed against Go 1.7). But that might be addressed in your followup PR. Thanks!

**Comment by dsymonds** [comment]
Yeah, I can't see why it is failing. I can see it works with Go 1.7, Go 1.9 and Go 1.10, just not Go 1.8. It's only the `TestClassifier_WithinConfidenceThreshold` test that fails, but not in a way that is obviously related to Go 1.8.

**Comment by gwelymernans** [comment]
*nods* It might be best just to ignore Go 1.8, since we're already two versions beyond it now...

**Pull request merged**